packages/editor-preview/src/browser/editor-preview-manager.ts

```diff
@@ -14,21 +14,41 @@
 // SPDX-License-Identifier: EPL-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0
 // *****************************************************************************
 
-import { EditorManager, EditorOpenerOptions, EditorWidget } from '@theia/editor/lib/browser';
-import { inject, injectable, postConstruct } from '@theia/core/shared/inversify';
+import { EditorManager, EditorOpenerOptions, EditorWidget, Range } from '@theia/editor/lib/browser';
+import { inject, injectable, named, postConstruct } from '@theia/core/shared/inversify';
 import { EditorPreviewPreferences } from './editor-preview-preferences';
-import { MaybePromise } from '@theia/core/lib/common';
+import { ContributionProvider, MaybePromise, Prioritizeable, RecursivePartial } from '@theia/core/lib/common';
 import URI from '@theia/core/lib/common/uri';
 import { EditorPreviewWidgetFactory, EditorPreviewOptions } from './editor-preview-widget-factory';
 import { EditorPreviewWidget } from './editor-preview-widget';
 import { FrontendApplicationStateService } from '@theia/core/lib/browser/frontend-application-state';
 
+export const EditorSelectionResolver = Symbol('EditorSelectionResolver');
+
+/**
+ * Resolves an initial selection to be revealed in an editor.
+ *
+ * Implementations may provide a custom initial selection based on the given widget, opener options,
+ * and optional URI. The resolver's priority determines the order of execution, with higher values executed first.
+ *
+ * @see EditorPreviewManager#revealSelection
+ */
+export interface EditorSelectionResolver {
+    /**
+     * The priority of the resolver. A higher value resolver will be called before others.
+     */
+    priority?: number;
+    resolveSelection(widget: EditorWidget, options: EditorOpenerOptions, uri?: URI): Promise<RecursivePartial<Range> | undefined>
+}
+
 @injectable()
 export class EditorPreviewManager extends EditorManager {
     override readonly id = EditorPreviewWidgetFactory.ID;
 
     @inject(EditorPreviewPreferences) protected readonly preferences: EditorPreviewPreferences;
     @inject(FrontendApplicationStateService) protected readonly stateService: FrontendApplicationStateService;
+    @inject(ContributionProvider) @named(EditorSelectionResolver)
```

Comment:
The "modern" style of implementing contribution is to have an API for dynamic contribution like 
```
registerSelectionResolver(priority: number, resolver: SelectionResolver): Disposable;
```
And then to register the contributions in a `@postConstruct` method. This allows for dynamic contributions, for example from a plugin API.